domains/authorization/Authorization.API/API.UnitTests/API.UnitTests/Repository/FakeAffilicationRepository.cs

```diff
@@ -0,0 +1,64 @@
+using API.Authorization.Exceptions;
+using API.Models;
+using API.Repository;
+using MockQueryable.NSubstitute;
+
+namespace API.UnitTests.Repository;
+
+public class FakeAffiliationRepository : IAffiliationRepository
+{
+    private readonly List<Affiliation> _entities = [];
+
+    public Task<Affiliation> GetAsync(object[] keys, CancellationToken cancellationToken)
+    {
+        var organisationId = (Guid)keys[0];
+        var userId = (Guid)keys[1];
+
+        var entity = _entities.Find(e => e.OrganizationId == organisationId && e.UserId == userId);
+        if (entity is null)
+        {
+            throw new EntityNotFoundException(organisationId + userId.ToString(), nameof(Affiliation));
+        }
+        return Task.FromResult(entity);
+    }
+
+    public Task<Affiliation> GetAsync(Guid id, CancellationToken cancellationToken)
+    {
+        throw new NotImplementedException();
```

Comment:
This perhaps shows that generic repository, doesn't make sense?

Reply:
removed repo